app/src/main/kotlin/com/intive/picover/auth/repository/AuthRepository.kt

```diff
@@ -41,19 +42,18 @@ class AuthRepository @Inject constructor(
 		firebaseAuth.signOut()
 	}
 
-	suspend fun userProfile(): Profile {
-		val photoUrl = userAvatarReference.downloadUrl.await()
-		return requireUser().let {
-			Profile(
-				photo = photoUrl,
-				name = it.displayName!!,
-				email = it.email!!,
-			)
+	suspend fun userProfile() =
+		try {
+			val photoUrl = userAvatarReference.downloadUrl.await()
+			Result.success(createProfile(photoUrl))
+		} catch (storageException: StorageException) {
+			Result.success(createProfile())
```

Comment:
I would pass explicitly `null` here to `createProfile()` function. Now it's a little bit hidden that we return a profile without uri in case of exception

Reply:
Done

Reply:
Not really, I still see the old version